cpp/serve/engine_actions/batch_draft.cc

```diff
@@ -0,0 +1,213 @@
+/*!
+ *  Copyright (c) 2023 by Contributors
+ * \file serve/engine_actions/batch_spec_decode.cc
+ */
+
+#include "../config.h"
+#include "../model.h"
+#include "../sampler.h"
+#include "action.h"
+#include "action_commons.h"
+
+namespace mlc {
+namespace llm {
+namespace serve {
+
+/*!
+ * \brief The action that runs draft proposal for requests in the
+ * `running_queue` of engine state. Preempt low-priority requests
+ * accordingly when it is impossible to decode all the running requests.
+ */
+class BatchDraftActionObj : public EngineActionObj {
+ public:
+  explicit BatchDraftActionObj(Array<Model> models, Sampler sampler,
+                               Optional<EventTraceRecorder> trace_recorder, int draft_length = 4)
```

Comment:
No need to have the default value here, given we provide the default value at an outer scope.
```suggestion
                               Optional<EventTraceRecorder> trace_recorder, int draft_length)
```